### ISSUE TITLE
Improve distributed tests for Confusion Matrix (CM and MLCM)

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#1657]
Currently, the testing data is generated without taking into account distributed context:
https://github.com/pytorch/ignite/blob/e17acc740d38909ddbbed6648423750abe593e58/tests/ignite/metrics/test_confusion_matrix.py#L559-L565

same for MultiLabelConfusionMatrix tests.

We would like to rewrite the tests in a similar way as here : 
https://github.com/pytorch/ignite/blob/e17acc740d38909ddbbed6648423750abe593e58/tests/ignite/metrics/test_recall.py#L778-L785

cc @touqir14

**Comment by Moh-Yakoub** [comment]
I saw this issue and couple of PRs linked to it. Is this fully done already?

**Comment by vfdev-5** [comment]
No it is still to be done

**Comment by touqir14** [comment]
I can start working on this but I am a bit busy at the moment and not before 4-5 days unfortunately. If @Moh-Yakoub wants to take a shot, that is fine.

**Comment by Moh-Yakoub** [comment]
I think I can start working on this.

**Comment by Moh-Yakoub** [comment]
@vfdev-5  I feel I am missing something here about the tests. Based on investigating the test cases in the confusion matrix file. I found that the distributed context was already based in the distributed test cases

For example:

`def test_distrib_gpu(local_rank, distributed_context_single_node_nccl):

    device = torch.device(f"cuda:{local_rank}")
    _test_distrib_multiclass_images(device)
    _test_distrib_accumulator_device(device)`

**Comment by vfdev-5** [comment]
@Moh-Yakoub yes, there are distributed tests, it is what I say in the issue description. The problem is that the data is generated without taking into account distributed context. Currently, the data is the same for all procs. What we want is to generate data dependant of the proc rank. Please, reread issue description for details.

**Comment by nafiulkhalid** [comment]
Hi! I'm Nafiul, a Computer Science senior at FIU, and I'd like to work on this issue **if it's still available**.

I reviewed the issue and the related distributed tests. My understanding is that the goal is to **make the test data generation depend on the distributed context**, similar to the distributed Recall tests.

I'll start by reproducing the current tests and reviewing the existing implementation. Please let me know if there are any additional considerations before I get started.

Thank you!

**Comment by nafiulkhalid** [comment]
Hi @vfdev-5, just following up on my comment from a few days ago. I'd like to work on this — I've started looking into the distributed test improvements for CM and MLCM. Let me know if there's a preferred approach or any context I should be aware of before I open a PR. Happy to proceed if this is still up for contribution!
